testing/routes.md

```diff
@@ -0,0 +1,129 @@
+We can test server routes using [Supertest](https://github.com/visionmedia/supertest) and [tape](https://github.com/substack/tape).
+
+
+### Setting up
+
+We do this first and foremost by _exporting_ the Express `app` object from `server.js` (or wherever it happens to be defined):
+
+```
+var app = express()
+
+// ...
+
+module.exports = app
+```
+
+Importantly, we want to put the `app.listen()` call in another file (`index.js`, for example). This allows our testing code to `require('./server.js')`. If we put the `.listen()` call in `server.js`, it will execute when the file is read and we don't actually want the server listening for connections while we're running our tests. This implies that we can tell things about our application without running it at all! Handy.
+
+`index.js` can be pretty simple:
+
+```
+var app = require('./server')
```

Comment:
To match the previous code snippet, I was thinking we could remove line 18 and add `// index.js` here. These are _far_ from significant changes so go with whatever you think is best.